GVRf/Framework/framework/src/main/jni/engine/renderer/batch.cpp

```diff
@@ -164,68 +160,52 @@ bool Batch::isRenderDataDisabled(){
     }
     return update_vbo;
 }
-void Batch::setupMesh(){
-    bool update_vbo = isRenderDataDisabled();
-
-    if(render_data_set_.size() ==0)
-        return;
-
-    if(update_vbo)
-        regenerateMeshData();
+void Batch::resetBatch(){
+    clearData();
+    delete renderdata_;
+    Renderer::batch_manager->deallocateBatch(this);
```

Comment:
You may consider a more canonical name for a pool like "free". deallocate could create a wrong impression.